src/Configuration.h

```diff
@@ -84,6 +85,7 @@ class Configuration{
     transform_loop_unroll = -1;
     print_progress = false;
     print_progress_estimate = false;
+    program_arguments = std::vector<std::string>();
```

Comment:
There are two test suite failures in the debug (compile against a debug build of llvm, and pass `--enable-assertions` to `./configure`) build:
```
./unittest --report_level=short --log_level=warning --run_test='*'
Running 534 test cases...
unittest: /home/magnus/repos/llvm/lib/ExecutionEngine/ExecutionEngine.cpp:456: int llvm::ExecutionEngine::runFunctionAsMain(llvm::Function*, const std::vector<std::__cxx11::basic_string<char> >&, const char* const*): Assertion `!isTargetNullPtr(this, GVTOP(GVArgs[1])) && "argv[0] was null after CreateArgv"' failed.
unknown location(0): fatal error: in "POWER_test/Main_arg": signal: SIGABRT (application abort requested)
../../src/POWER_test2.cpp(887): last checkpoint: "Main_arg" test entry
unittest: /home/magnus/repos/llvm/lib/ExecutionEngine/ExecutionEngine.cpp:456: int llvm::ExecutionEngine::runFunctionAsMain(llvm::Function*, const std::vector<std::__cxx11::basic_string<char> >&, const char* const*): Assertion `!isTargetNullPtr(this, GVTOP(GVArgs[1])) && "argv[0] was null after CreateArgv"' failed.
unknown location(0): fatal error: in "POWER_test/Assertion_abort": signal: SIGABRT (application abort requested)
../../src/POWER_test2.cpp(903): last checkpoint: "Assertion_abort" test entry
```
What I think's happening is that since `Configuration::assign_by_commandline()` is not called when running the testsuite, `conf.program_arguments` stay the empty vector, and thus the test programs get `argc=0`, which is not a valid value for `argc`. Probably, if you just put `"prog"` in here instead, the failures will go away.

Reply:
Nice catch. I will implement the fix that you suggest.